org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/popup/AbstractNotificationPopup.java

```diff
@@ -0,0 +1,567 @@
+
+    /* Create Title Area */
+    createTitleArea(titleCircle);
+
+    /* Outer composite to hold content controlls */
+    Composite outerContentCircle = new Composite(outerCircle, SWT.NONE);
+    outerContentCircle.setBackgroundMode(SWT.INHERIT_FORCE);
+
+    layout = new GridLayout(1, false);
+    layout.marginWidth = 0;
+    layout.marginHeight = 0;
+
+    outerContentCircle.setLayout(layout);
+    outerContentCircle.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+    outerContentCircle.setBackground(outerCircle.getBackground());
+
+    /* Middle composite to show a 1px black line around the content controls */
+    Composite middleContentCircle = new Composite(outerContentCircle, SWT.NO_FOCUS);
+    middleContentCircle.setBackgroundMode(SWT.INHERIT_FORCE);
+
+    layout = new GridLayout(1, false);
+    layout.marginWidth = 0;
+    layout.marginHeight = 0;
+    layout.marginTop = 1;
+
+    middleContentCircle.setLayout(layout);
+    middleContentCircle.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+    middleContentCircle.setBackground(color.getBorder());
+
+    /* Inner composite containing the content controls */
+    Composite innerContent = new Composite(middleContentCircle, SWT.NO_FOCUS);
+    innerContent.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+    innerContent.setBackgroundMode(SWT.INHERIT_FORCE);
+
+    layout = new GridLayout(1, false);
+    layout.marginWidth = 0;
+    layout.marginHeight = 5;
+    layout.marginLeft = 5;
+    layout.marginRight = 5;
+    innerContent.setLayout(layout);
+
+    innerContent.setBackground(shell.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+
+    /* Content Area */
+    createContentArea(innerContent);
+
+    setNullBackground(outerCircle);
+
+    return outerCircle;
+  }
+
+  private void setNullBackground(final Composite outerCircle) {
+    for (Control c : outerCircle.getChildren()) {
+      c.setBackground(null);
+      if (c instanceof Composite) {
+        setNullBackground((Composite) c);
+      }
+    }
+  }
+
+  @Override
+  protected void initializeBounds() {
+    Rectangle clArea = getPrimaryClientArea();
+    Point initialSize = shell.computeSize(SWT.DEFAULT, SWT.DEFAULT);
+    int height = Math.max(initialSize.y, MIN_HEIGHT);
+    int width = Math.min(initialSize.x, MAX_WIDTH);
+
+    Point size = new Point(width, height);
+    shell.setLocation(clArea.width + clArea.x - size.x - PADDING_EDGE, clArea.height + clArea.y - size.y
+      - PADDING_EDGE);
+    shell.setSize(size);
+  }
+
+  private Rectangle getPrimaryClientArea() {
+    Monitor primaryMonitor = shell.getDisplay().getPrimaryMonitor();
+    return (primaryMonitor != null) ? primaryMonitor.getClientArea() : shell.getDisplay().getClientArea();
+  }
+
+  public void closeFade() {
+    if (fadeJob != null) {
+      fadeJob.cancelAndWait(false);
+    }
+    fadeJob = AnimationUtil.fadeOut(getShell(), new IFadeListener() {
+      public void faded(Shell shell, int alpha) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1161)